### ISSUE TITLE
Optuna tuning gives worse results than my randomly chosen hyperparameters.

### ISSUE THREAD

**Issue by beast331** [optuna/optuna#3318]
### Environment

- Optuna version: 2.10.0
- Python version: 3.9.7
- OS: Windows-10-10.0.22000-SP0
- (Optional) Other libraries and their versions: LightGBM Version 3.3.2

### Description

Hi everyone. I have a dataset with 10 million rows and 26 features. I am using LightGBM to make predictions with evaluation metric as rmse. My randomly chosen parameters with a bit of manual experimentation is as follows:

```python
params = {}
params['learning_rate'] = 0.01
params['boosting_type'] = 'gbdt'
params['metric'] = 'rmse'
params['max_depth'] = 12
params['num_leaves'] = 2**7
params['min_data_in_bin'] = 20
params['objective'] = 'regression'
params['feature_fraction'] = 0.9
params['bagging_fraction'] = 0.8
params['bagging_freq'] = 5
params['num_threads'] = 4
params['device'] = 'gpu'
params['verbose'] = 2
```

My rmse score was capping at about 1.18 on the test set with these parameters, so I decided to use Optuna to do hyperparameter tuning. As the dataset is time-series based, I decided to use TimeSeriesSplit for cross validation. My code is as follows.

```python
import optuna
from sklearn.model_selection import StratifiedKFold
import lightgbm as lgbm

def objective(trial, X, y):
    
    param_grid = {
        #"device_type": trial.suggest_categorical("device_type", ['gpu']),
        "n_estimators": trial.suggest_categorical("n_estimators", [10000]),
        "learning_rate": trial.suggest_float("learning_rate", 0.01, 0.3),
        "num_leaves": trial.suggest_int("num_leaves", 20, 3000, step=20),
        "max_depth": trial.suggest_int("max_depth", 3, 12),
        "min_data_in_leaf": trial.suggest_int("min_data_in_leaf", 200, 10000, step=100),
        "max_bin": trial.suggest_int("max_bin", 200, 300),
        "lambda_l1": trial.suggest_int("lambda_l1", 0, 100, step=5),
        "lambda_l2": trial.suggest_int("lambda_l2", 0, 100, step=5),
        "min_gain_to_split": trial.suggest_float("min_gain_to_split", 0, 15),
        "bagging_fraction": trial.suggest_float(
            "bagging_fraction", 0.2, 0.95, step=0.1
        ),
        "bagging_freq": trial.suggest_categorical("bagging_freq", [1]),
        "feature_fraction": trial.suggest_float(
            "feature_fraction", 0.2, 0.95, step=0.1
        ),
    }
    
    
    
    cv = TimeSeriesSplit(gap=0, max_train_size=None, n_splits=5, test_size=None)

    cv_scores = np.empty(5)
    count = 0
    for idx, (train_idx, test_idx) in enumerate(cv.split(X, y)):
        count+= 1
        print("trial %d"%(count))
        X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
        y_train, y_test = y[train_idx], y[test_idx]

        model = lgbm.LGBMRegressor(objective="regression", **param_grid)
        model.fit(
            X_train,
            y_train,
            eval_set=[(X_test, y_test)],
            eval_metric="rmse",
            early_stopping_rounds=100
        )
        preds = model.predict(X_test)
        cv_scores[idx] = math.sqrt(mean_squared_error(y_test, preds))

    return np.mean(cv_scores)

study = optuna.create_study(direction="minimize", study_name="LGBM Classifier")
func = lambda trial: objective(trial, X, y)
study.optimize(func, n_trials=20)
```

After running 20 trials, the following hyperparameters were chosen as best for my model. 

```
Best value (rmse): 0.85001
	Best params:
		n_estimators: 10000
		learning_rate: 0.2805603007218991
		num_leaves: 560
		max_depth: 9
		min_data_in_leaf: 2600
		max_bin: 264
		lambda_l1: 100
		lambda_l2: 0
		min_gain_to_split: 1.2545283531220668
		bagging_fraction: 0.6000000000000001
		bagging_freq: 1
		feature_fraction: 0.8
```

However, after running this model and uploading the predictions to Kaggle, my score has worsened to 1.7. Is there any reason for this? Am I making an error somewhere. Thank you!

### Error messages, stack traces, or logs

```shell
NA
```

### Steps to reproduce

1. 
2.
3.
```python
# python code
```

**Comment by nzw0301** [comment]
Let me point out a few things to improve the performance.

1. The searched space of a few hyperparameters in `objective` does not cover mutually set hyperparameters. For example `bagging_freq`. If the manually selected parameters are almost optimal, the searched place should cover them, ideally, not the minimum/max value. for example, `learnint_rage`
2. `n_trials` might be too small.
3. Could you compare the validation dataset instead of Kaggle's private data?

**Comment by beast331** [comment]
> Let me point out a few things to improve the performance.
> 
> 1. The searched space of a few hyperparameters in `objective` does not cover mutually set hyperparameters. For example `bagging_freq`. If the manually selected parameters are almost optimal, the searched place should cover them, ideally, not the minimum/max value. for example, `learnint_rage`
> 2. `n_trials` might be too small.
> 3. Could you compare the validation dataset instead of Kaggle's private data?

Hi @nzw0301 , thank you for your reply. I will attempt the same with 50 trials and see if I get any improvement. Also, I believe I made some errors in the way I implemented the algorithm, and actually with optuna parameters I managed to achieve a score of 1.17. Will further trials help to improve this score?

**Comment by nzw0301** [comment]
I'm afraid not

**Comment by nzw0301** [comment]
I believe that I've answered the original question, and the optimisation problem was not reproducible because the dataset was not publicly available and depends on some private data on Kaggle. So, let me close this issue.
